src/plugins/query_enhancements/public/query_assist/utils/get_is_summary_agent.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { HttpSetup } from 'opensearch-dashboards/public';
+import { API } from '../../../common/constants';
+
+export async function checkAgentsExist(
+  http: HttpSetup,
+  agentConfigName: string | string[],
+  dataSourceId: string
+) {
+  try {
+    const queryParams = new URLSearchParams();
+    if (Array.isArray(agentConfigName)) {
+      agentConfigName.forEach((name) => queryParams.append('agentConfigName', name));
+    } else {
+      queryParams.append('agentConfigName', agentConfigName);
+    }
+    queryParams.append('dataSourceId', dataSourceId);
+
+    const response = await http.get(API.AGENT_API.CONFIG_EXISTS, {
+      query: Object.fromEntries(queryParams),
+    });
+    return response;
+  } catch (error) {
+    throw error;
```

Comment:
btw, I don't see the purpose of this `catch` because it throws it immediately which is the same as not having the try-catch block.

Reply:
ok, I see. thank you